internal/internal_workflow.go

```diff
@@ -1347,3 +1356,55 @@ func (h *queryHandler) execute(input []byte) (result []byte, err error) {
 	}
 	return result, err
 }
+
+// Add adds delta, which may be negative, to the WaitGroup counter.
+// If the counter becomes zero, all goroutines blocked on Wait are released.
+// If the counter goes negative, Add panics.
+//
+// Note that calls with a positive delta that occur when the counter is zero
+// must happen before a Wait. Calls with a negative delta, or calls with a
+// positive delta that start when the counter is greater than zero, may happen
+// at any time.
+// Typically this means the calls to Add should execute before the statement
+// creating the goroutine or other event to be waited for.
+// If a WaitGroup is reused to wait for several independent sets of events,
+// new Add calls must happen after all previous Wait calls have returned.
+//
+// param delta int32 -> the value to increment the WaitGroup counter by
+func (wg *waitGroupImpl) Add(delta int32) {
```

Comment:
Any specific reason the delta is int32? The sync.WaitGroup.Add is int.

Reply:
If I remember correctly, it looked like in sync.WaitGroup() were only using 32 bits to store the counter. I am storing the counter as an int32 and I wanted that to be consistent with Add()

Reply:
This is the WaitGroup signature:

`func (wg *WaitGroup) Add(delta int) {
`
I think we want to have exactly the same API

Reply:
OK I'll make the change.